vault/dynamic_system_view.go

```diff
@@ -0,0 +1,54 @@
+package vault
+
+import (
+	"fmt"
+	"strings"
+	"time"
+)
+
+type dynamicSystemView struct {
+	core *Core
+	path string
```

Comment:
Can we just put the *MountEntry in here directly and thus avoid the call to `d.core.router.MatchingMountEntry`?